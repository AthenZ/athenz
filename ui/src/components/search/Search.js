/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { Router } from '../../routes';
import SearchInput from '../denali/SearchInput';

export default class Search extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            selected: this.props.option || 'domain',
            searchText: this.props.searchData ? this.props.searchData : '',
        };
    }

    render() {
        return (
            <SearchInput
                dark={!!this.props.isHeader}
                name='search'
                fluid={true}
                value={this.state.searchText}
                placeholder='Enter domain name'
                error={this.state.error}
                onChange={(event) =>
                    this.setState({
                        searchText: event.target.value,
                        error: false,
                    })
                }
                onKeyPress={(event) => {
                    if (event.key === 'Enter') {
                        if (this.state.searchText.length === 0) {
                            this.setState({
                                error: true,
                            });
                        } else {
                            Router.pushRoute('search', {
                                type: this.state.selected,
                                searchterm: this.state.searchText.trim(),
                            });
                        }
                    }
                }}
            />
        );
    }
}
