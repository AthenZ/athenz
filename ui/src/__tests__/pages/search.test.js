/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { render } from '@testing-library/react';
import Search from '../../pages/search';

describe('Search', () => {
    it('should render', () => {
        let domains = [];
        domains.push({ name: 'athens' });
        domains.push({ name: 'athens.ci' });

        let result = {
            name: 'test1',
            adminDomain: true,
        };
        let domainResults = [];
        domainResults.push(result);
        let headerDetails = {
            headerLinks: [
                {
                    title: 'Website',
                    url: 'http://www.athenz.io',
                    target: '_blank',
                },
            ],
        };

        const { getByTestId } = render(
            <Search
                domains={domains}
                domain='test'
                type='domain'
                domainResults={domainResults}
                userId={'pgote'}
                headerDetails={headerDetails}
            />
        );
        const search = getByTestId('search');
        expect(search).toMatchSnapshot();
    });
});
