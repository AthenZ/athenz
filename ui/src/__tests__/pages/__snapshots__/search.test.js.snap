// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render 1`] = `
.emotion-11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #002339;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  box-sizing: border-box;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 60px;
  -ms-flex: 0 0 60px;
  flex: 0 0 60px;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  height: 60px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  left: 0;
  padding: 0 24px;
  position: static;
  right: 0;
  top: 0;
  z-index: 2;
}

.emotion-11.scrolled {
  box-shadow: 0px 4px 8px rgba(0,0,0,0.16);
}

.emotion-11 .flush-right {
  margin-left: auto;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-1.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-1.simple.nav-link {
  cursor: pointer;
}

.emotion-1.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-1.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-1.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-1 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-1.nav-link .nav-title {
  cursor: pointer;
}

.emotion-1.active .nav-title,
.emotion-1 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-1 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-1.nav-link.active .nav-title:hover,
.emotion-1.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-1 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-1 .nav-icon,
.emotion-1:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-1 .nav-icon path,
.emotion-1:hover .nav-icon path,
.emotion-1 .nav-icon ellipse,
.emotion-1:hover .nav-icon ellipse,
.emotion-1 .nav-icon rect,
.emotion-1:hover .nav-icon rect,
.emotion-1 .nav-icon circle,
.emotion-1:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-1.nav-link:hover .nav-icon path,
.emotion-1.nav-link:hover .nav-icon ellipse,
.emotion-1.nav-link:hover .nav-icon rect,
.emotion-1.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
  width: 100%;
}

.emotion-4.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-4.simple.nav-link {
  cursor: pointer;
}

.emotion-4.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-4.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-4.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-4 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-4.nav-link .nav-title {
  cursor: pointer;
}

.emotion-4.active .nav-title,
.emotion-4 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-4 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-4.nav-link.active .nav-title:hover,
.emotion-4.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-4 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-4 .nav-icon,
.emotion-4:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-4 .nav-icon path,
.emotion-4:hover .nav-icon path,
.emotion-4 .nav-icon ellipse,
.emotion-4:hover .nav-icon ellipse,
.emotion-4 .nav-icon rect,
.emotion-4:hover .nav-icon rect,
.emotion-4 .nav-icon circle,
.emotion-4:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-4.nav-link:hover .nav-icon path,
.emotion-4.nav-link:hover .nav-icon ellipse,
.emotion-4.nav-link:hover .nav-icon rect,
.emotion-4.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  position: relative;
  width: 100%;
}

.emotion-3 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: rgba(53,112,244,0.05);
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-right: transparent;
  border-left: transparent;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color: #303030;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font: inherit;
  height: 36px;
  margin: 0;
  outline: none;
  padding: 0 1rem;
  text-align: left;
  width: 100%;
}

.emotion-3.animated input {
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out, border 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out, border 0.2s ease-in-out;
}

.emotion-3 input::-webkit-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-3 input::-moz-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-3 input:-ms-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-3 input::placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-3 input.focused,
.emotion-3 input:active,
.emotion-3 input:focus {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #3570f4;
  color: #303030;
}

.emotion-3 input:invalid {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #d01111;
  color: #303030;
}

.emotion-3 .message {
  font-size: 12px;
  left: 0;
  line-height: 1.8;
  position: absolute;
  top: 2.571rem;
}

.emotion-3.error input,
.emotion-3.error input.focused,
.emotion-3.error input:active,
.emotion-3.error input:focus {
  border-bottom: 2px solid #d01111;
}

.emotion-3.error .message {
  color: #d01111;
}

.emotion-3 input {
  background: rgba(242,242,242,0.1);
  color: #fff;
}

.emotion-3 input::-webkit-input-placeholder {
  color: rgba(255,255,255,0.6);
}

.emotion-3 input::-moz-placeholder {
  color: rgba(255,255,255,0.6);
}

.emotion-3 input:-ms-input-placeholder {
  color: rgba(255,255,255,0.6);
}

.emotion-3 input::placeholder {
  color: rgba(255,255,255,0.6);
}

.emotion-3 input.focused,
.emotion-3 input:active,
.emotion-3 input:focus {
  background: rgba(242,242,242,0.1);
  border-bottom: 2px solid #3570f4;
  color: #fff;
}

.emotion-3 input {
  padding-right: 36px;
}

.emotion-3 .input-icon {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 36px;
  position: absolute;
  right: 6px;
}

.emotion-2 {
  fill: #3570f4;
  cursor: inherit;
  vertical-align: text-bottom;
}

.emotion-6 {
  fill: #fff;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-17 {
  fill: #303030;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-23 {
  fill: #303030;
  cursor: inherit;
  vertical-align: text-bottom;
}

.emotion-30 {
  fill: #303030;
  cursor: inherit;
  vertical-align: baseline;
}

.emotion-12 {
  height: 60px;
  position: relative;
}

.emotion-0 {
  height: 36px;
  cursor: pointer;
}

.emotion-5 {
  margin-left: 10%;
  width: 50%;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  margin-left: 15px;
}

.emotion-41 {
  -webkit-flex: 1 1 calc(100vh - 60px);
  -ms-flex: 1 1 calc(100vh - 60px);
  flex: 1 1 calc(100vh - 60px);
  overflow: hidden;
  font: 300 14px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
}

.emotion-40 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-22 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  height: calc(100vh - 60px);
  overflow: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-16 {
  background: #f5f8fe;
  padding: 20px 30px 0;
}

.emotion-13 {
  font: 600 20px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  margin-bottom: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-14 {
  color: #9a9a9a;
}

.emotion-15 {
  border-bottom: 1px solid #d5d5d5;
  margin-top: 10px;
  margin-bottom: 10px;
  width: 100%;
}

.emotion-20 {
  color: #3570f4;
  margin-bottom: 5px;
  font: 100 14px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  cursor: pointer;
  padding: 0 30px 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-18 {
  font-size: 1.25em;
  margin-right: 5px;
  vertical-align: text-bottom;
}

.emotion-19 {
  color: #3570f4;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-24 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-bottom: 1px solid #d5d5d5;
  border-left: 1px solid #d5d5d5;
  border-top: 1px solid #d5d5d5;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: -21px;
  margin-top: 20px;
  position: absolute;
  text-align: center;
  width: 20px;
}

.emotion-39 {
  margin-right: 0;
  border-left: 1px solid #d5d5d5;
  -webkit-flex: 0 0 350px;
  -ms-flex: 0 0 350px;
  flex: 0 0 350px;
  height: calc(100vh - 60px);
  overflow: auto;
  -webkit-transition: margin 0.4s ease-in-out;
  transition: margin 0.4s ease-in-out;
  display: block;
  min-width: 350px;
  width: 350px;
}

.emotion-29 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px 30px 20px 15px;
}

.emotion-25 {
  font-size: 16px;
  font-weight: 600;
}

.emotion-26 {
  color: #3570f4;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-27 {
  padding: 0 5px;
  color: #d5d5d5;
}

.emotion-38 {
  padding: 0 30px 0 15px;
}

.emotion-33 {
  padding: 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-31 {
  font-size: 1.25em;
  margin-right: 5px;
}

<div
  data-testid="search"
>
  <div
    class="emotion-12"
    data-testid="header"
  >
    <div
      class="emotion-11 denali-navbar"
      data-testid="navbar"
    >
      <div
        class="emotion-1 simple denali-navbar-item"
        data-testid="navbar-item"
      >
        <a
          href="/"
        >
          <img
            class="emotion-0"
            src="/static/athenz-logo-full.png"
          />
        </a>
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-4 simple denali-navbar-item"
          data-testid="navbar-item"
        >
          <div
            class="emotion-3 denali-input animated"
            data-testid="input-wrapper"
          >
            <input
              autocomplete="off"
              data-testid="input-node"
              name="search"
              placeholder="Enter domain name"
              value="test"
            />
            <div
              class="input-icon"
            >
              <svg
                class="emotion-2"
                data-testid="icon"
                height="24px"
                viewBox="0 0 1024 1024"
                width="24px"
              >
                <title>
                  search
                </title>
                <path
                  d="M659.413 599.68c40.125-53.054 64.27-120.136 64.27-192.857 0-177.556-143.937-321.493-321.493-321.493s-321.493 143.937-321.493 321.493c0 177.556 143.937 321.493 321.493 321.493 74.297 0 142.708-25.203 197.149-67.525l-0.726 0.543 267.307 264.96c7.723 7.731 18.396 12.514 30.187 12.514s22.464-4.782 30.186-12.513l0-0c7.795-7.733 12.621-18.45 12.621-30.293s-4.826-22.56-12.618-30.291l-0.003-0.003zM444.8 637.227c-11.829 2.113-25.446 3.322-39.345 3.322-129.603 0-234.667-105.064-234.667-234.667s105.064-234.667 234.667-234.667c129.603 0 234.667 105.064 234.667 234.667 0 13.899-1.208 27.516-3.526 40.752l0.204-1.407c-17.178 98.349-93.651 174.822-190.593 191.796l-1.407 0.204z"
                />
              </svg>
            </div>
            <div
              class="message"
              data-testid="message"
            />
          </div>
        </div>
      </div>
      <div
        class="emotion-1 simple flush-right denali-navbar-item"
        data-testid="navbar-item"
      >
        <div
          class="emotion-9"
          data-testid="header-menu"
        >
          <svg
            class="emotion-6"
            data-testid="icon"
            height="25px"
            viewBox="0 0 1024 1024"
            width="25px"
          >
            <title>
              notification
            </title>
            <path
              d="M925.867 627.2l-68.267-68.267c-15.132-16.256-24.706-37.864-25.595-61.692l-0.005-0.174v-134.4c0-172.8-115.2-320-320-320s-320 142.933-320 320v149.333c-0.424 14.075-6.029 26.765-14.962 36.297l0.028-0.030-78.933 78.933c-7.763 7.612-12.628 18.155-12.8 29.835l-0 0.032v153.6c0 23.564 19.103 42.667 42.667 42.667v0h236.8c12.454 70.953 73.611 124.182 147.2 124.182s134.746-53.229 147.069-123.285l0.131-0.897h236.8c23.564 0 42.667-19.103 42.667-42.667v0-153.6c-0.172-11.712-5.037-22.255-12.793-29.86l-0.007-0.007zM853.333 768h-682.667v-91.733l66.133-68.267c24.782-24.511 40.224-58.423 40.533-95.941l0-0.059v-149.333c0-113.067 61.867-234.667 234.667-234.667s234.667 125.867 234.667 234.667v134.4c0.879 47.11 19.381 89.736 49.168 121.71l-0.102-0.11 57.6 57.6z"
            />
          </svg>
          <div
            class="emotion-8"
          >
            <svg
              class="emotion-6"
              data-testid="icon"
              height="25px"
              viewBox="0 0 1024 1024"
              width="25px"
            >
              <title>
                help-circle
              </title>
              <path
                d="M512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0-259.206-210.128-469.333-469.333-469.333v0zM512 896c-212.077 0-384-171.923-384-384s171.923-384 384-384c212.077 0 384 171.923 384 384v0c0 212.077-171.923 384-384 384v0z"
              />
              <path
                d="M554.667 768c0 23.564-19.103 42.667-42.667 42.667s-42.667-19.103-42.667-42.667c0-23.564 19.103-42.667 42.667-42.667s42.667 19.103 42.667 42.667z"
              />
              <path
                d="M549.333 219.52c-11.244-2.339-24.166-3.678-37.4-3.678-71.49 0-133.854 39.072-166.904 97.027l-0.496 0.945c-3.783 6.395-6.018 14.093-6.018 22.313 0 14.468 6.924 27.319 17.64 35.419l0.112 0.081c6.829 4.846 15.334 7.747 24.517 7.747 15.995 0 29.934-8.801 37.239-21.826l0.111-0.215c12.111-22.947 31.869-40.492 55.815-49.471l0.718-0.236c98.133-33.707 187.52 72.32 120.96 170.667-16.346 22.524-42.587 37.002-72.21 37.002-4.025 0-7.988-0.267-11.871-0.785l0.455 0.050c-1.281-0.156-2.763-0.245-4.267-0.245s-2.986 0.089-4.443 0.262l0.176-0.017c-19.477 4.079-33.927 21.016-34.133 41.364l-0 0.022s0 0 0 1.28v82.773c-0.004 0.232-0.007 0.505-0.007 0.78 0 21.42 15.032 39.329 35.123 43.752l0.297 0.055c2.182 0.396 4.694 0.622 7.258 0.622 23.564 0 42.667-19.103 42.667-42.667 0-0.219-0.002-0.437-0.005-0.655l0 0.033v-31.147c0.196-9.48 6.547-17.425 15.209-20.015l0.151-0.039c78.046-25.561 133.424-97.737 133.424-182.847 0-92.861-65.924-170.325-153.523-188.144l-1.234-0.21z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-41"
  >
    <div
      class="emotion-40"
    >
      <div
        class="emotion-22"
      >
        <div
          class="emotion-21"
        >
          <div
            class="emotion-16"
          >
            <div
              class="emotion-13"
            >
              Search Results
            </div>
            <div
              class="emotion-14"
            >
              1
               Results
            </div>
            <div
              class="emotion-15"
            />
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-18"
            >
              <svg
                class="emotion-17"
                data-testid="icon"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <title>
                  user-secure
                </title>
                <path
                  d="M512 42.667s-384 108.8-384 128v388.267c0 196.267 283.733 369.067 362.667 416 5.993 4.054 13.381 6.471 21.333 6.471s15.341-2.418 21.469-6.558l-0.136 0.087c81.067-44.8 362.667-219.733 362.667-416v-388.267c0-19.2-384-128-384-128zM512 887.467c-67.707-40.17-126.321-83.163-180.386-131.167l1.186 1.034c53.263-19.828 114.792-31.517 178.988-31.999l0.212-0.001c64.373 0.253 125.934 11.964 182.856 33.195l-3.656-1.195c-52.498 48.488-111.237 91.596-174.507 127.669l-4.693 2.464zM810.667 558.933c0 34.133-17.067 78.933-55.467 130.133-71.851-30.233-155.31-48.181-242.855-49.064l-0.345-0.003c-87.136 0.665-169.896 18.64-245.256 50.648l4.19-1.581c-40.533-51.2-57.6-96-57.6-130.133v-334.933c61.867-23.467 185.6-59.733 298.667-91.733 113.067 32 236.8 70.4 298.667 91.733z"
                />
                <path
                  d="M512 298.667c47.128 0 85.333 38.205 85.333 85.333s-38.205 85.333-85.333 85.333c-47.128 0-85.333-38.205-85.333-85.333v0c0-47.128 38.205-85.333 85.333-85.333v0zM512 213.333c-94.257 0-170.667 76.41-170.667 170.667s76.41 170.667 170.667 170.667c94.257 0 170.667-76.41 170.667-170.667v0c0-94.257-76.41-170.667-170.667-170.667v0z"
                />
              </svg>
            </div>
            <a
              class="emotion-19"
            >
              test1
            </a>
          </div>
        </div>
      </div>
      <div
        data-testid="user-domains"
      >
        <div
          class="emotion-24"
          data-testid="toggle-domain"
        >
          <svg
            class="emotion-23"
            data-testid="icon"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <title>
              arrow-right
            </title>
            <path
              d="M579.627 512l-225.707 226.133c-7.425 7.668-12.002 18.133-12.002 29.667 0 14.028 6.77 26.476 17.222 34.252l0.114 0.081c6.877 4.635 15.35 7.398 24.467 7.398 12.618 0 24-5.292 32.048-13.778l0.019-0.020 254.293-253.653c7.731-7.723 12.514-18.396 12.514-30.187s-4.782-22.464-12.513-30.186l-0-0-254.293-254.507c-8.066-8.506-19.448-13.798-32.066-13.798-9.118 0-17.59 2.763-24.625 7.498l0.158-0.1c-10.919 7.83-17.949 20.483-17.949 34.778 0 11.841 4.824 22.556 12.613 30.286l0.003 0.003z"
            />
          </svg>
        </div>
        <div
          class="emotion-39"
        >
          <div
            class="emotion-29"
          >
            <div
              class="emotion-25"
            >
              My Domains
            </div>
            <div>
              <a
                class="emotion-26"
              >
                Create
              </a>
              <span
                class="emotion-27"
              >
                 | 
              </span>
              <a
                class="emotion-26"
              >
                Manage
              </a>
            </div>
          </div>
          <div
            class="emotion-38"
          >
            <div
              class="emotion-33"
            >
              <div
                class="emotion-31"
              >
                <svg
                  class="emotion-30"
                  data-testid="icon"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <title>
                    user-group
                  </title>
                  <path
                    d="M320 512c106.039 0 192-85.961 192-192s-85.961-192-192-192c-106.039 0-192 85.961-192 192v0c0 106.039 85.961 192 192 192v0zM320 213.333c58.91 0 106.667 47.756 106.667 106.667s-47.756 106.667-106.667 106.667c-58.91 0-106.667-47.756-106.667-106.667v0c0-58.91 47.756-106.667 106.667-106.667v0z"
                  />
                  <path
                    d="M746.667 512c82.475 0 149.333-66.859 149.333-149.333s-66.859-149.333-149.333-149.333c-82.475 0-149.333 66.859-149.333 149.333v0c0 82.475 66.859 149.333 149.333 149.333v0zM746.667 298.667c35.346 0 64 28.654 64 64s-28.654 64-64 64c-35.346 0-64-28.654-64-64v0c-0.024-0.647-0.038-1.408-0.038-2.171 0-34.168 27.699-61.867 61.867-61.867 0.764 0 1.524 0.014 2.281 0.041l-0.11-0.003z"
                  />
                  <path
                    d="M981.333 627.2l-19.2-12.8c-61.484-37.554-135.879-59.79-215.467-59.79s-153.983 22.236-217.31 60.834l1.843-1.044-17.067 12.8c-51.435-18.731-110.808-29.656-172.706-29.866l-0.094-0c-1.041-0.008-2.273-0.012-3.505-0.012-102.805 0-198.352 31.076-277.759 84.348l1.797-1.135-19.2 12.8v160c0 23.564 19.103 42.667 42.667 42.667v0h512c23.564 0 42.667-19.103 42.667-42.667v0-42.667h298.667c23.564 0 42.667-19.103 42.667-42.667v0zM554.667 810.667h-426.667v-70.4c60.429-36.199 133.319-57.605 211.209-57.605 0.747 0 1.493 0.002 2.239 0.006l-0.115-0c0.631-0.003 1.377-0.005 2.124-0.005 77.89 0 150.78 21.406 213.108 58.657l-1.899-1.052zM896 725.333h-256v-32l-32-21.333c40.883-19.81 88.92-31.389 139.661-31.389 54.141 0 105.204 13.182 150.153 36.512l-1.814-0.857z"
                  />
                </svg>
              </div>
              <a
                class="emotion-26"
              >
                athens
              </a>
            </div>
            <div
              class="emotion-33"
            >
              <div
                class="emotion-31"
              >
                <svg
                  class="emotion-30"
                  data-testid="icon"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <title>
                    user-group
                  </title>
                  <path
                    d="M320 512c106.039 0 192-85.961 192-192s-85.961-192-192-192c-106.039 0-192 85.961-192 192v0c0 106.039 85.961 192 192 192v0zM320 213.333c58.91 0 106.667 47.756 106.667 106.667s-47.756 106.667-106.667 106.667c-58.91 0-106.667-47.756-106.667-106.667v0c0-58.91 47.756-106.667 106.667-106.667v0z"
                  />
                  <path
                    d="M746.667 512c82.475 0 149.333-66.859 149.333-149.333s-66.859-149.333-149.333-149.333c-82.475 0-149.333 66.859-149.333 149.333v0c0 82.475 66.859 149.333 149.333 149.333v0zM746.667 298.667c35.346 0 64 28.654 64 64s-28.654 64-64 64c-35.346 0-64-28.654-64-64v0c-0.024-0.647-0.038-1.408-0.038-2.171 0-34.168 27.699-61.867 61.867-61.867 0.764 0 1.524 0.014 2.281 0.041l-0.11-0.003z"
                  />
                  <path
                    d="M981.333 627.2l-19.2-12.8c-61.484-37.554-135.879-59.79-215.467-59.79s-153.983 22.236-217.31 60.834l1.843-1.044-17.067 12.8c-51.435-18.731-110.808-29.656-172.706-29.866l-0.094-0c-1.041-0.008-2.273-0.012-3.505-0.012-102.805 0-198.352 31.076-277.759 84.348l1.797-1.135-19.2 12.8v160c0 23.564 19.103 42.667 42.667 42.667v0h512c23.564 0 42.667-19.103 42.667-42.667v0-42.667h298.667c23.564 0 42.667-19.103 42.667-42.667v0zM554.667 810.667h-426.667v-70.4c60.429-36.199 133.319-57.605 211.209-57.605 0.747 0 1.493 0.002 2.239 0.006l-0.115-0c0.631-0.003 1.377-0.005 2.124-0.005 77.89 0 150.78 21.406 213.108 58.657l-1.899-1.052zM896 725.333h-256v-32l-32-21.333c40.883-19.81 88.92-31.389 139.661-31.389 54.141 0 105.204 13.182 150.153 36.512l-1.814-0.857z"
                  />
                </svg>
              </div>
              <a
                class="emotion-26"
              >
                athens.ci
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
